api/pkg/s3/bucketdeleteCORS.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/emicklei/go-restful"
+	"github.com/micro/go-log"
+	"github.com/opensds/multi-cloud/api/pkg/common"
+	s3 "github.com/opensds/multi-cloud/s3/proto"
+)
+
+func (s *APIService) BucketDeleteCORS(request *restful.Request, response *restful.Response) {
+
+	ctx := common.InitCtxWithAuthInfo(request)
+	//var foundID int
+	FoundIDArray := []string{}
+	NonFoundIDArray := []string{}
+	bucketName := request.PathParameter("bucketName")
+	corsID := request.Request.URL.Query()["corsID"]
+	if corsID != nil {
+		bucket, _ := s.s3Client.GetBucket(ctx, &s3.Bucket{Name: bucketName})
+		for _, id := range corsID {
+			isfound := false
+			for _, lcRule := range bucket.CorsConfiguration {
+				if lcRule.Id == id {
+					isfound = true
+					FoundIDArray = append(FoundIDArray, id)
+				}
```

Comment:
For optimization, you can add break statement here, so that once "id" found further for loop should not execute